src/script/components/InputBar/index.tsx

```diff
@@ -861,20 +871,70 @@ const InputBar = ({
                     suggestions={mentionSuggestions}
                     onSelectionValidated={addMention}
                   />
-
-                  <InputBarControls
-                    conversation={conversationEntity}
-                    input={inputValue}
-                    isEditing={isEditing}
-                    disablePing={pingDisabled}
-                    disableFilesharing={!isFileSharingSendingEnabled}
-                    onSend={() => onSend(inputValue)}
-                    onSelectFiles={uploadFiles}
-                    onSelectImages={uploadImages}
-                    onCancelEditing={cancelMessageEditing}
-                    onClickGif={onGifClick}
-                    onClickPing={onPingClick}
-                  />
+                  {isScaledDown ? (
+                    <>
+                      <ul className="controls-right buttons-group" css={{minWidth: '95px'}}>
+                        {showGiphyButton && <GiphyButton onGifClick={onGifClick} />}
+                        <li>
+                          <button
+                            type="button"
+                            className={cx('controls-right-button controls-right-button--send')}
+                            disabled={inputValue.length === 0}
+                            title={t('tooltipConversationSendMessage')}
+                            aria-label={t('tooltipConversationSendMessage')}
+                            onClick={() => onSend(inputValue)}
+                            data-uie-name="do-send-message"
+                          >
+                            <Icon.Send />
+                          </button>
+                        </li>
+                      </ul>
+                      <ul className="controls-right buttons-group" css={{justifyContent: 'center', width: '100%'}}>
+                        <ControlButtons
+                          conversation={conversationEntity}
+                          input={inputValue}
+                          isEditing={isEditing}
+                          isScaledDown={isScaledDown}
+                          disablePing={pingDisabled}
+                          disableFilesharing={!isFileSharingSendingEnabled}
+                          onSelectFiles={uploadFiles}
+                          onSelectImages={uploadImages}
+                          onCancelEditing={cancelMessageEditing}
+                          onGifClick={onGifClick}
+                          onClickPing={onPingClick}
```

Comment:
Quite a bit of duplicated code between the 2 conditions. 
let's try to extract the component instantiation to variables and then place them according to the layout. 

something like:

```jsx
const sendButton = <button ...><Icon.Send /></button>;
const controlButtons = <ControlButtons ... />

return ...
isScaledDown ? <ul>{sendButton}</ul><ul>{controlButtons}</ul> : <ul>{controlButtons}</ul><ul>{sendButton></ul>
```

Reply:
got something like this:

```
  const controlButtonsProps = {
    conversation: conversationEntity,
    disableFilesharing: isFileSharingSendingEnabled,
    disablePing: pingDisabled,
    input: inputValue,
    isEditing: isEditing,
    onCancelEditing: cancelMessageEditing,
    onClickPing: onPingClick,
    onGifClick: onGifClick,
    onSelectFiles: uploadFiles,
    onSelectImages: uploadImages,
  };
```

```
                    <ul className="controls-right buttons-group">
                      <ControlButtons {...controlButtonsProps} showGiphyButton={showGiphyButton} />
                      {sendButton}
                    </ul>
```